packages/eclwatch/src/ECLArchiveViewer.ts

```diff
@@ -0,0 +1,343 @@
+    folderIconClosed(): string;
+    folderIconClosed(_: string): this;
+    hoverBackgroundColor(): string;
+    hoverBackgroundColor(_: string): this;
+    rowItemPadding(): number;
+    rowItemPadding(_: number): this;
+    codeFileIcon(): string;
+    codeFileIcon(_: string): this;
+    textFileIcon(): string;
+    textFileIcon(_: string): this;
+    verticalScroll(): boolean;
+    verticalScroll(_: boolean): this;
+}
+ArchiveList.prototype.publish("rowItemPadding", 2, "number", "Top, bottom, left and right row item padding");
+ArchiveList.prototype.publish("hoverBackgroundColor", "#CCC", "html-color", "Background color of hovered directory rows");
+ArchiveList.prototype.publish("backgroundColor", "#FFF", "html-color", "Directory list background color");
+ArchiveList.prototype.publish("fontColor", "#000", "html-color", "Directory list font color");
+ArchiveList.prototype.publish("fontFamily", "Arial", "string", "Directory list font family");
+ArchiveList.prototype.publish("fontSize", 12, "number", "Directory list font family (pixels)");
+ArchiveList.prototype.publish("iconSize", 12, "number", "Directory folder and file icon size (pixels)");
+ArchiveList.prototype.publish("folderIconOpen", "fa fa-folder-open", "string", "Open folder icon class");
+ArchiveList.prototype.publish("folderIconClosed", "fa fa-folder", "string", "Closed folder icon class");
+ArchiveList.prototype.publish("textFileIcon", "fa fa-file-text-o", "string", "Text file icon class");
+ArchiveList.prototype.publish("codeFileIcon", "fa fa-file-code-o", "string", "Code file icon class");
+ArchiveList.prototype.publish("verticalScroll", true, "boolean", "If true, vertical scroll bar will be shown");
+
+export class ECLArchiveViewer extends SplitPanel {
+    public _directoryPane = new ArchiveList();
+    public _fileEditorPane = new Editor()
+        .text("")
+        ;
+
+    constructor() {
+        super("horizontal");
+        this._directoryPane.rowClick = (text) => {
+            this._fileEditorPane
+                .text(text)
+                .render()
+                ;
+        };
+    }
+    enter(domNode, element) {
+        super.enter(domNode, element);
+        this
+            .addWidget(this._directoryPane)
+            .addWidget(this._fileEditorPane)
+            ;
+    }
+    update(domNode, element) {
+        super.update(domNode, element);
+
+        let json;
+        const contentStr = this.contentString();
+        if (contentStr) {
+            try {
+                json = JSON.parse(contentStr);
+            } catch (e) {
+                json = xml2json(contentStr);
+            }
+        }
+        if (json) {
+            const _data = this.transformArchiveTreeData(json);
+            this._directoryPane
+                .data(_data)
+                .render()
+                ;
+        }
+        this.relativeSizes([this.directoryWidthRatio(), 1 - this.directoryWidthRatio()]);
+    }
+
+    transformArchiveTreeData(json) {
```

Comment:
For consistency move to above enter/update/exit methods
...and mark as protected